torch/csrc/jit/python/script_init.cpp

```diff
@@ -783,11 +783,46 @@ void initJitScriptBindings(PyObject* module) {
                   if (auto method = self.find_method(name)) {
                     return py::cast(*method);
                   }
+                  if (self.has_property(name)) {
+                    auto prop = self.get_property(name);
+                    auto getter_func = py::cast(prop.getter_func);
+                    return getter_func();
+                  }
                   return toPyObject(self.attr(name));
                 } catch (const ObjectAttributeError& err) {
                   throw AttributeError("%s", err.what());
                 }
               })
+          .def(
+              "__setattr__",
+              [](Object& self, const std::string& name, py::object value) {
+                try {
+                  if (self.has_property(name)) {
+                    auto prop = self.get_property(name);
+                    if (!prop.setter_func.has_value()) {
+                      TORCH_CHECK(false, "can't set attribute");
+                    }
+                    auto setter_func = py::cast(prop.setter_func);
+                    setter_func(value);
```

Comment:
Consider adding a comment explaining why `prop.setter_func` isn't called directly.

Potential follow-up: calling `invokeScriptMethodFromPython` here directly instead of using the pybind type for `Method`.